core/modules.rs

```diff
@@ -210,11 +210,15 @@ pub trait ModuleLoader {
   ///
   /// `is_main` can be used to resolve from current working directory or
   /// apply import map for child imports.
+  ///
+  /// `is_dyn_import` can be used to check permissions or deny
+  /// dynamic imports altogether.
   fn resolve(
     &self,
     specifier: &str,
     referrer: &str,
     _is_main: bool,
+    _is_dynamic: bool,
```

Comment:
```suggestion
    _is_main: bool,
    _is_dynamic: bool,
```

Maybe this is something for a follow up, but I would really prefer if these were enums rather than plain booleans. Makes it more obvious what is being passed around and what each state means.

Reply:
Added `deno_core::ResolutionKind` enum to solve this.